### PR TITLE
Add procedures for incremental exports

### PR DESCRIPTION
**Pull request by bangelic**
New procedures were required for incremental exports and have been
created for the Managing Content Guide. This is a pull request taking
over for #2049. 

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.6/Katello 4.8
* [x] Foreman 3.5/Katello 4.7 (planned Satellite 6.13)
* [x] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only)
* [ ] Foreman 3.2/Katello 4.4 on EL7 & EL8
* [ ] Foreman 3.1/Katello 4.3 on EL7 & EL8 (Satellite 6.11 EL7/8, orcharhino 6.2 on EL7/8)
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.4 or older.

### CODE REVIEW

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental repository \
+ --organization="_My_Organization_" \
+ --product="_My_Product_" \
+ --name="_My_Repository_" \
+ --format=syncable
+Generated /var/lib/pulp/exports/Default_Organization/Export-SYNCABLE-Red_Hat_Ansible_Engine_2_for_RHEL_8_x86_64_RPMs-1/2.0/2023-03-09T10-55-48-05-00/metadata.json
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:22]:
```suggestion
```

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+----
+. Optional: View the exported data:
++
+[options="nowrap" subs="+quotes"]
+----
+# cd /var/lib/pulp/exports/Default_Organization/Export-SYNCABLE-Red_Hat_Ansible_Engine_2_for_RHEL_8_x86_64_RPMs-1/2.0/2023-03-09T10-55-48-05-00/
+# find . -name *.rpm
+./content/dist/layered/rhel8/x86_64/ansible/2/os/Packages/s/sshpass-1.06-3.el8ae.x86_64.rpm
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:30]:
```suggestion
# find /var/lib/pulp/exports/Default_Organization/Export-SYNCABLE-Red_Hat_Ansible_Engine_2_for_RHEL_8_x86_64_RPMs-1/2.0/2023-03-09T10-55-48-05-00/ -name "*.rpm"
```

If you're worried about showing a very long path for each package, you could add `| xargs basename -a` which strips with path and only shows the RPM file name.

File: guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[id="Exporting_Library_Incrementally_in_a_Syncable_Format_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc:1]:
```suggestion
[id="Exporting_the_Library_Environment_Incrementally_in_a_Syncable_Format_{context}"]
```

File: guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+The procedure below shows incremental export of all repositories in the organization's Library.
+
+.Procedure
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental library --organization="_My_Organization_" --format=syncable
+Generated /var/lib/pulp/exports/Default_Organization/Export-Library-SYNCABLE/2.0/2023-03-09T11-00-21-05-00/metadata.json
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc:21]:
```suggestion
# hammer content-export incremental library \
--format=syncable \
--organization="_My_Organization_"
```

File: guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+----
+. Optional: View the exported data:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc:23]: same suggestion as above.

File: guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+----
+# hammer content-export incremental library \
+--format=syncable \
+--organization="_My_Organization_"
+----
+. Optional: View the exported data:
++
+[options="nowrap" subs="+quotes"]
+----
+# find /var/lib/pulp/exports/Default_Organization/Export-SYNCABLE-Red_Hat_Ansible_Engine_2_for_RHEL_8_x86_64_RPMs-1/2.0/2023-03-09T10-55-48-05-00/ -name "*.rpm"
```
Comment (vsedmik) [inline guides/common/modules/proc_exporting-the-library-environment-incrementally-in-a-syncable-format.adoc:28]:
```suggestion
# find /var/lib/pulp/exports/Default_Organization/Export-Library-SYNCABLE/2.0/2023-03-09T10-55-48-05-00/ -name "*.rpm"
```
The export path for `library` is different than `product` or `CVV`

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+# hammer content-export incremental version \
+--content-view="_Content_View_Name_" \
+--organization="_My_Organization_" \
+--version=2.0 --format=syncable
+----
+. Optional: View the exported Content View:
++
+[options="nowrap" subs="+quotes"]
+----
+# cd /var/lib/pulp/exports/Default_Organization/view/3.0/2023-03-02T14-19-50-05-00/
```
Comment (vsedmik) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:29]:
```suggestion
# cd /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/
```
above we exported `--version=2.0`

**Review by vsedmik — APPROVED** [review]
Looks good to me, I just added two nitpick comments.

File: guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc [hunk 8]
```diff
@@ -1,7 +1,10 @@
 [id="Exporting_a_Content_View_Version_in_a_Syncable_Format_{context}"]
 = Exporting a Content View Version in a Syncable Format
 
-You can export a version of a Content View to a syncable format that you can use to create your custom CDN and synchronize the content from the custom CDN over HTTP/HTTPS.
+You can export a version of a Content View to a syncable format that you can use to create your custom CDN. Once exported, do either of the following:
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc:4]:
```suggestion
You can export a version of a Content View to a syncable format that you can use to create your custom CDN.
After you have exported the Content View, you can do either of the following:
```

File: guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc [hunk 9]
```diff
@@ -1,7 +1,10 @@
...
+
+* Synchronize the content from the custom CDN over HTTP/HTTPS.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc:6]:
```suggestion
* Synchronize the content from your custom CDN over HTTP/HTTPS.
```

File: guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc [hunk 10]
```diff
@@ -1,7 +1,10 @@
...
+* Import the content via `hammer content-import` which is available if both the Export and Import servers are {ProductVersion} and above.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc:7]:
```suggestion
* Import the content using `hammer content-import`.
Note that this requires both the Export and Import servers are {ProductVersion} and above.
```

I am unsure about the second sentence because the docs are specifically built for {ProductVersion} anyway. Maybe reword so that users ensure both the export and import {ProjectServer} are on the same/target version?

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 11]
```diff
@@ -0,0 +1,32 @@
+[id="Exporting_Version_Incrementally_in_a_Syncable_Format_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:1]:
```suggestion
[id="Exporting_a_Content_View_Version_Incrementally_in_a_Syncable_Format_{context}"]
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+= Exporting a Content View Version Incrementally in a Syncable Format
+
+Exporting complete Content View versions can be a very costly operation in terms of system resources.
+ifdef::orcharhino[]
+The size of the exported Content View versions depends on the number of products.
+endif::[]
+Content View versions that have multiple {RHEL} trees can occupy several gigabytes of space on {ProjectServer}.
+
+In such cases, you can use *Incremental Export* to export only pieces of content that changed since the previous export.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:10]:
```suggestion
In such cases, you can create an incremental export which contains only pieces of content that changed since the previous export.
```

I am unsure if my suggestion is a good idea; but the highlighting for `*Incremental Export*` is not a button in the Web UI.

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+The example below targets the version `2.0` for export, because the version `1.0` was exported previously.
+
+.Procedure
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental version \
+--content-view="_Content_View_Name_" \
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:21]:
```suggestion
--content-view="_My_Content_View_Name_" \
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+--organization="_My_Organization_" \
+--version=2.0 --format=syncable
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:23]:
```suggestion
--version=2.0 \
--format=syncable
```

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 15]
```diff
@@ -0,0 +1,28 @@
+[id="Exporting_Repository_Incrementally_in_a_Syncable_Format_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:1]:
```suggestion
[id="Exporting_a_Repository_Incrementally_in_a_Syncable_Format_{context}"]
```

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+= Exporting a Repository Incrementally in a Syncable Format
+
+Exporting a repository can be a very costly operation in terms of system resources.
+A typical {RHEL} tree may occupy several gigabytes of space on {ProjectServer}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:5]: This is missing the "orcharhino only" sentence from above.

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+ --organization="_My_Organization_" \
+ --product="_My_Product_" \
+ --name="_My_Repository_" \
+ --format=syncable
+----
+. Optional: View the exported data:
++
+[options="nowrap" subs="+quotes"]
+----
+# find /var/lib/pulp/exports/Default_Organization/Export-SYNCABLE-Red_Hat_Ansible_Engine_2_for_RHEL_8_x86_64_RPMs-1/2.0/2023-03-09T10-55-48-05-00/ -name "*.rpm"
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:27]: I like this code block better than the one above. Maybe use this one above as well instead of "cd && find"?

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc (hunk 16 above)
Comment (bangelic) [reply guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:5]: @maximiliankolb Which sentence is that?

Comment (maximiliankolb) [reply guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:5]:
````
ifdef::orcharhino[]
The size of the exported Content View versions depends on the number of products.
endif::[]
````

See guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc in your PR.

**Comment by bangelic** [comment]
The build keeps breaking because of a cross reference that is being added to 6.13.

File: guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc [hunk 18]
```diff
@@ -1,7 +1,12 @@
 [id="Exporting_a_Content_View_Version_in_a_Syncable_Format_{context}"]
 = Exporting a Content View Version in a Syncable Format
 
-You can export a version of a Content View to a syncable format that you can use to create your custom CDN and synchronize the content from the custom CDN over HTTP/HTTPS.
+You can export a version of a Content View to a syncable format that you can use to create your custom CDN.
+After you have exported the Content View, you can do either of the following:
+
+* Synchronize the content from your custom CDN over HTTP/HTTPS.
+* Import the content using `hammer content-import`.
+Note that this requires both the Export and Import servers are the same {ProductVersion} and above.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc:9]: This sounds off to me. Please reword.

I am also unsure if "current version or above" make sense in different branches.

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc [hunk 19]
```diff
@@ -1,4 +1,4 @@
-[id="Exporting_Version_Incrementally_{context}"]
+[id="Exporting_a_Content_View_Version_Incrementally_in_a_Syncable_Format_{context}"]
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:1]: Anchor does not match title and file name.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
Made some small suggestions; rest LGTM

File: guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc [hunk 20]
```diff
@@ -1,7 +1,12 @@
 [id="Exporting_a_Content_View_Version_in_a_Syncable_Format_{context}"]
 = Exporting a Content View Version in a Syncable Format
 
-You can export a version of a Content View to a syncable format that you can use to create your custom CDN and synchronize the content from the custom CDN over HTTP/HTTPS.
+You can export a version of a Content View to a syncable format that you can use to create your custom CDN.
+After you have exported the Content View, you can do either of the following:
+
+* Synchronize the content from your custom CDN over HTTP/HTTPS.
+* Import the content using `hammer content-import`.
+Note that this requires both the Export and Import servers are the same {ProductVersion}.
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-in-a-syncable-format.adoc:9]:
```suggestion
Note that this requires both the Export and Import servers to run {Project} {ProductVersion}.
```

File: guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc [hunk 21]
```diff
@@ -0,0 +1,28 @@
...
+.Procedure
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental repository \
+ --organization="_My_Organization_" \
+ --product="_My_Product_" \
+ --name="_My_Repository_" \
+ --format=syncable
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-repository-incrementally-in-a-syncable-format.adoc:21]:
```suggestion
--organization="_My_Organization_" \
--product="_My_Product_" \
--name="_My_Repository_" \
--format=syncable
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+----
+. Optional: View the exported Content View:
++
+[options="nowrap" subs="+quotes"]
+----
+# cd /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/
+# find . -name *.rpm
+./content/dist/layered/rhel8/x86_64/ansible/2/os/Packages/s/sshpass-1.06-3.el8ae.x86_64.rpm
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:31]:
```suggestion
# find /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/ -name "*.{pkg_ext}"
```

**Review by maximiliankolb — COMMENTED** [review]
some more minor thing, rest LGTM!

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 23]
```diff
@@ -0,0 +1,29 @@
+[id="Exporting_a_Content_View_Version_Incrementally_in_a_Syncable_Format_{context}"]
+= Exporting a Content View Version Incrementally in a Syncable Format
+
+Exporting complete Content View versions can be a very costly operation in terms of system resources.
+ifdef::orcharhino[]
+The size of the exported Content View versions depends on the number of products.
+endif::[]
+Content View versions that have multiple {RHEL} trees can occupy several gigabytes of space on {ProjectServer}.
+
+In such cases, you can use *Incremental Export* to export only pieces of content that changed since the previous export.
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:10]:
```suggestion
In such cases, you can use *Incremental Export* to export only pieces of content that got changed since the previous export.
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc [hunk 24]
```diff
@@ -7,7 +7,7 @@ The size of the exported Content View versions depends on the number of products
 endif::[]
 Content View versions that have multiple {RHEL} trees can occupy several gigabytes of space on {ProjectServer}.
 
-In such cases, you can use *Incremental Export* to export only pieces of content that changed since the previous export.
+In such cases, you can create an incremental export which contains only pieces of content that changed since the previous export.
```
Comment (mjivraja) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:10]:
```suggestion
In such cases, you can create an incremental export which contains only pieces of content that got changed since the previous export.
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc (hunk 23 above)
Comment (bangelic) [reply guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:10]: I think it's probably best to leave as is so it doesn't sound past tense with `got`.

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc (hunk 24 above)
Comment (bangelic) [reply guides/common/modules/proc_exporting-a-content-view-version-incrementally.adoc:10]: Same as above, I think it's probably best to leave as is so it doesn't sound past tense with `got`.

**Comment by bangelic** [comment]
@maximiliankolb Is this good to be merged?

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+The example below targets the version `2.0` for export because the version `1.0` was exported previously.
+
+.Procedure
+. Create an incremental export:
++
+[options="nowrap" subs="+quotes"]
+----
+# hammer content-export incremental version \
+--content-view="_Content_View_Name_" \
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:21]:
```suggestion
--content-view="_My_Content_View_Name_" \
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+--organization="_My_Organization_" \
+--version=2.0 --format=syncable
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:23]:
```suggestion
--version=2.0 \
--format=syncable
```

File: guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+----
+. Optional: View the exported Content View:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# find /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/ -name "*.{pkg_ext}"----
```
Comment (maximiliankolb) [inline guides/common/modules/proc_exporting-a-content-view-version-incrementally-in-a-syncable-format.adoc:29]:
```suggestion
# find /var/lib/pulp/exports/Default_Organization/view/2.0/2023-03-02T14-19-50-05-00/ -name "*.{pkg_ext}"
----
```

this suggestions hopefully fixes the broken build.

**Review by maximiliankolb — CHANGES_REQUESTED** [review]
made some comments which hopefully fix the broken build. rest LGTM.

**Comment by bangelic** [comment]
Thanks @maximiliankolb. Build looks good!

**Review by maximiliankolb — APPROVED** [review]
LGTM!

**Comment by maximiliankolb** [comment]
Merged to master and cherry-picked: :heavy_check_mark: 
   66b15975e..29796d0b8  3.6 -> 3.6
   4c697da00..d338cb8a9  3.5 -> 3.5
   9505a67e7..5d9eaa5d0  3.4 -> 3.4
